pkg/kubelet/cm/memorymanager/policy_static_test.go

```diff
@@ -2340,3 +2344,1724 @@ func TestStaticPolicyGetTopologyHints(t *testing.T) {
+						},
+						hugepages1Gi: {
+							Allocatable:    10 * gb,
+							Free:           10 * gb,
+							Reserved:       0,
+							SystemReserved: 0,
+							TotalMemSize:   10 * gb,
+						},
+					},
+					Nodes:               []int{1},
+					NumberOfAssignments: 0,
+				},
+			},
+			expectedMachineState: state.NodeMap{
+				0: &state.NodeState{
+					MemoryMap: map[v1.ResourceName]*state.MemoryTable{
+						v1.ResourceMemory: {
+							Allocatable:    9 * gb,
+							Free:           4 * gb,
+							Reserved:       5 * gb,
+							SystemReserved: 1 * gb,
+							TotalMemSize:   10 * gb,
+						},
+						hugepages1Gi: {
+							Allocatable:    10 * gb,
+							Free:           5 * gb,
+							Reserved:       5 * gb,
+							SystemReserved: 0,
+							TotalMemSize:   10 * gb,
+						},
+					},
+					Nodes:               []int{0},
+					NumberOfAssignments: 4,
+				},
+				1: &state.NodeState{
+					MemoryMap: map[v1.ResourceName]*state.MemoryTable{
+						v1.ResourceMemory: {
+							Allocatable:    9 * gb,
+							Free:           9 * gb,
+							Reserved:       0,
+							SystemReserved: 1 * gb,
+							TotalMemSize:   10 * gb,
+						},
+						hugepages1Gi: {
+							Allocatable:    10 * gb,
+							Free:           10 * gb,
+							Reserved:       0,
+							SystemReserved: 0,
+							TotalMemSize:   10 * gb,
+						},
+					},
+					Nodes:               []int{1},
+					NumberOfAssignments: 0,
+				},
+			},
+			pod: getPod("pod2", "container2", requirementsGuaranteed),
+			systemReserved: systemReservedMemory{
+				0: map[v1.ResourceName]uint64{
+					v1.ResourceMemory: 1 * gb,
+				},
+				1: map[v1.ResourceName]uint64{
+					v1.ResourceMemory: 1 * gb,
+				},
+			},
+			expectedTopologyHints: map[string][]topologymanager.TopologyHint{
+				string(v1.ResourceMemory): {
+					{
+						NUMANodeAffinity: newNUMAAffinity(0),
+						Preferred:        true,
+					},
+				},
+				string(hugepages1Gi): {
+					{
+						NUMANodeAffinity: newNUMAAffinity(0),
+						Preferred:        true,
+					},
+				},
+			},
+			memoryToReuse: map[string]map[v1.ResourceName]state.Block{
+				"pod2": map[v1.ResourceName]state.Block{
+					"memory": state.Block{
+						NUMAAffinity: []int{0},
+						Type:         v1.ResourceMemory,
+						Size:         5 * gb,
+					},
+					hugepages1Gi: state.Block{
+						NUMAAffinity: []int{0},
+						Type:         hugepages1Gi,
+						Size:         5 * gb,
+					},
+				},
+			},
+			expectedMemoryToReuse: map[string]map[v1.ResourceName]state.Block{
+				"pod2": map[v1.ResourceName]state.Block{
+					"memory": state.Block{
+						NUMAAffinity: []int{0},
+						Type:         v1.ResourceMemory,
+						Size:         4 * gb,
+					},
+					hugepages1Gi: state.Block{
+						NUMAAffinity: []int{0},
+						Type:         hugepages1Gi,
+						Size:         4 * gb,
+					},
+				},
+			},
+		},
+		{
+			description: "app containers request higher than reusable, should allocate from reusable and from nodemap",
```

Comment:
Does it the same test as https://github.com/mgjeong/kubernetes/pull/30/files#diff-7ddae4e94b2576ee01e6b475370eb488a637d80bbe6f391ae6cdfd128311811aR3109?

Reply:
They were different:
- in first init container has hugepages that can be reused
- in second it doesn't 
I updated descriptions to be more precise. If you think that one of those tests is redundant I can delete it. (I wonder if it is)